src/routes/transactions/mappers/common/transaction.mapper.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@nestjs/common';
+import { Transaction as TransactionDomain } from '../../../../domain/safe/entities/transaction.entity';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { ModuleTransactionMapper } from '../module-transactions/module-transaction.mapper';
+import { MultisigTransactionMapper } from '../multisig-transactions/multisig-transaction.mapper';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import { EthereumTransaction } from '../../../../domain/safe/entities/ethereum-transaction.entity';
+import { flatten } from 'lodash';
+import { IncomingTransferMapper } from '../transfers/transfer.mapper';
+import { TransactionHistory } from '../../entities/transaction-history.entity';
+
+@Injectable()
+export class TransactionMapper {
+  constructor(
+    private readonly multisigTransactionMapper: MultisigTransactionMapper,
+    private readonly moduleTransactionMapper: ModuleTransactionMapper,
+    private readonly incomingTransferMapper: IncomingTransferMapper,
+  ) {}
+  async mapTransaction(
+    chainId: string,
+    transactionsDomain: TransactionDomain[],
+    safe: Safe,
+  ): Promise<TransactionHistory[]> {
+    const results = await transactionsDomain.map(async (transaction) => {
+      if ('isExecuted' in transaction) {
```

Comment:
We should work with domain entities at this level and use type predicates instead.

Reply:
This should be helpful to your PR. I'm fine extracting this to a different PR.

https://github.com/5afe/safe-client-gateway-nest/pull/247/files#diff-b5a64dd573f3ba03d25e2ac0a50dd52bc7935a0f217c0c2f2984a375a35c6579R10-R26

Reply:
Yes it is, but I guess that your PR is going to be merged soon, I can wait and see if there is any place more that we can be aligned.